File: 2000_reverse_prefix_of_word/reversePrefix2.py
class Solution:
    def reversePrefix(self, word: str, char: str) -> str:
        charIndex = word.find(char)
        res = ''
        l, r = 0, charIndex
        while l <= r:
            res += word[r]
            r-=1
        res +=  word[charIndex + 1:]
       
        return res

File: 2000_reverse_prefix_of_word/reversePrefix2.java
class Solution {
    public String reversePrefix(String word, char ch) {
        int charIndex = word.indexOf(ch);

        String prefix = reverse(word.substring(0, charIndex + 1));
        String postfix = word.substring(charIndex + 1);

        return prefix + postfix;
    }

    private String reverse(String word) {
        String res = "";

        for (int i = word.length() - 1; i >= 0; i--) {
            res += word.charAt(i);
        }

        return res;
    }
}